### PR TITLE
Particle property Solid Comp 

### PR DESCRIPTION
**Pull request by joeschools**
I added a new particle property solid comp which records the peridotite depletion. I believe this is currently the best way in aspect to track the petrological evolution of the the model and in conjunction with pT path, can be used to compare to MELTS or other thermodynamic software/databases.

### CODE REVIEW

**Comment by joeschools** [comment]
Property renamed as requested @gassmoeller

**Comment by joeschools** [comment]
Changes made. Now tracks all composition fields.

File: doc/modules/changes/20170515_schools [hunk 1]
```diff
@@ -1,6 +1,8 @@
 New: A new particle property is added which indicates the presence of melt 
 greater than the melt transport threshold at the particles position. If melt 
 is not present a 0 is recorded. If melt is present a 1 is recorded. Only works
-if there is a compositonal field named "porosity".
+if there is a compositonal field named "porosity". A second new particle property
+is also added which indicates the peridotite depletion value. This can be used
+to track the petrological evolution of material.
```
Comment (gassmoeller) [inline doc/modules/changes/20170515_schools]: Please update the entry.

File: source/particle/property/composition.cc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        std::vector<std::pair<std::string,unsigned int> > property_information;
+
+
+
+        for (unsigned int i = 0; i < this->n_compositional_fields(); i++)
+          {
+            std::ostringstream field_name;
+            field_name << this->introspection().name_for_compositional_index(i);
+            property_information.push_back(std::make_pair(field_name.str(),1));
```
Comment (gassmoeller) [inline source/particle/property/composition.cc]: You can simplify these 3 lines into the following:
```
const std::string field_name = this->introspection().name_for_compositional_index(i);
property_information.push_back(std::make_pair(field_name,1));
```

**Comment by joeschools** [comment]
Requested changes made. Thank you for suffering through my ignorance and thank you for all your help over the last ten days.

**Comment by gassmoeller** [comment]
The last formality. You will need to download the `changes-gcc.diff` file that you find when clicking on 'Details' next to the test results. Put it in your source folder and navigate in a terminal into that folder. Then apply the following patch:
```
patch -p1 < changes-gcc.diff
```
Make a new commit and push your branch. Then the tester should pass, and this is ready to go.

**Comment by joeschools** [comment]
Patch applied

**Comment by gassmoeller** [comment]
Thanks! Ready to go.

**Pull request merged**